hibernate-core/src/main/java/org/hibernate/type/descriptor/java/FloatJavaType.java

```diff
@@ -149,9 +154,14 @@ public long getDefaultSqlLength(Dialect dialect, JdbcType jdbcType) {
 
 	@Override
 	public int getDefaultSqlPrecision(Dialect dialect, JdbcType jdbcType) {
-		//this is the number of *binary* digits
-		//in a single-precision FP number
-		return dialect.getFloatPrecision();
+		if ( jdbcType.isFloat() ) {
+			//this is the number of *binary* digits
+			//in a single-precision FP number
+			return dialect.getFloatPrecision();
+		}
+		else {
+			return Math.min( dialect.getDefaultDecimalPrecision(), (int) ceil( dialect.getFloatPrecision() * LOG_BASE10OF2 ) );
```

Comment:
Similarly, should this just be 8 instead of using `dialect.getFloatPrecision()`?